test/integration/narrow/routes/agreement-submit/agreement-submit.test.js

```diff
@@ -0,0 +1,87 @@
+const db = require('../../../../../app/data')
+
+describe('agreement submit route', () => {
+  let createServer
+  let server
+  let agreementData
+  let progressData
+
+  beforeAll(async () => {
+    agreementData = {
+      agreementNumber: '123456789',
+      sbi: 123456789,
+      agreement: {}
+    }
+
+    progressData = {
+      progressId: 0,
+      progress: {
+        eligibility: true,
+        businessDetails: true
+      }
+    }
+
+    await db.agreement.destroy({ truncate: { cascade: true } })
+  })
+
+  beforeEach(async () => {
+    createServer = require('../../../../../app/server')
+    server = await createServer()
+    await server.initialize()
+  })
+
+  afterEach(async () => {
+    await server.stop()
+  })
+
+  afterAll(async () => {
+    await db.agreement.destroy({ truncate: { cascade: true } })
+    await db.sequelize.close()
+  })
+
+  test('POST /agreement/submit returns 201', async () => {
```

Comment:
For the names I'd maybe add some context to the test.

For example `Post xxx returns 201 when xxx` so it's clearer what the tests are checking